### PR TITLE
[OS-22]: Create user backend

### PR DESCRIPTION
**Pull request by uladzislau-zhurauski**

### CODE REVIEW

File: shop/controllers/user.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    def remove_username_unique_validator(cls, fields):
+        username_validators = fields.get('username').validators
+        for validator in username_validators:
+            if isinstance(validator, UniqueValidator):
+                username_validators.remove(validator)
+                break
+
+    @classmethod
+    def compute_superuser_settings(cls, user, is_staff, is_superuser, is_active):
+        if user.is_staff:
```
Comment (veranika-zakharchenia) [inline shop/controllers/user.py:56]: what is the purpose of passing user object and is_staff and compare below?

File: shop/controllers/user.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            cls.validate_superuser_settings(is_staff, is_superuser)
+        else:
+            is_staff = is_superuser = False
+            is_active = True
+        return is_staff, is_superuser, is_active
```
Comment (veranika-zakharchenia) [inline shop/controllers/user.py:61]: you need to create 3 separate methods to evaluate these parameters. It looks better than sending a tuple

File: shop/urls.py [hunk 3]
```diff
@@ -28,5 +29,10 @@
     path('orders/<int:pk>/', OrderView.as_view(), name='order-detail'),
     path('order-items/', OrderItemView.as_view(), name='order-item-list'),
     path('order-items/<int:pk>/', OrderItemView.as_view(), name='order-item-detail'),
+    path('users/', UserView.as_view(), name='user-list'),
+    path('users/<int:pk>/', UserView.as_view(), name='user-detail'),
+    path('users/<int:pk>/addresses/', UserAddressesView.as_view(), name='user-addresses'),
+    path('users/<int:pk>/feedback/', UserFeedbackView.as_view(), name='user-feedback'),
```
Comment (veranika-zakharchenia) [inline shop/urls.py:35]: feedbacks?

**Review by veranika-zakharchenia — CHANGES_REQUESTED** [review]
.

File: shop/controllers/user.py (hunk 1 above)
Comment (uladzislau-zhurauski) [reply shop/controllers/user.py:56]: Only staff users can set is_staff, is_superuser and is_active. This is the purpose of this method.

File: shop/urls.py (hunk 3 above)
Comment (uladzislau-zhurauski) [reply shop/urls.py:35]: Word  feedback have the same word for the plural.

**Review by veranika-zakharchenia — APPROVED** [review]

**Pull request merged**